MediaGalleryMetadata/Model/Writer/File.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryMetadata\Model\Writer;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\MediaGalleryMetadata\Model\File as FileDataObject;
+use Magento\MediaGalleryMetadata\Model\FileFactory;
+use Magento\MediaGalleryMetadata\Model\SegmentFactory;
+use Magento\MediaGalleryMetadata\Model\SegmentNames;
+
+/**
+ * File segments reader
+ */
+class File
+{
+    private const MARKER_IMAGE_FILE_START = "\xD8";
+    private const MARKER_IMAGE_PREFIX = "\xFF";
+    private const MARKER_IMAGE_END = "\xD9";
+
+    /**
+     * @var SegmentFactory
+     */
+    private $segmentFactory;
+
+    /**
+     * @var FileFactory
+     */
+    private $fileFactory;
+
+    /**
+     * @var SegmentNames
+     */
+    private $segmentNames;
+
+    /**
+     * @param FileFactory $fileFactory
+     * @param SegmentFactory $segmentFactory
+     * @param SegmentNames $segmentNames
+     */
+    public function __construct(FileFactory $fileFactory, SegmentFactory $segmentFactory, SegmentNames $segmentNames)
+    {
+        $this->fileFactory = $fileFactory;
+        $this->segmentFactory = $segmentFactory;
+        $this->segmentNames = $segmentNames;
+    }
+
+    /**
+     * @param FileDataObject $file
+     * @throws LocalizedException
+     */
+    public function execute(FileDataObject $file): void
+    {
+        foreach ($file->getSegments() as $segment) {
+            if (strlen($segment->getData()) > 0xfffd) {
```

Comment:
Do we need to use hexadecimal here, wouldn't decimal work as well?

Reply:
hexidecimal looks much more cool